gpt4all-chat/server.cpp

```diff
@@ -365,6 +369,8 @@ QHttpServerResponse Server::handleCompletionRequest(const QHttpServerRequest &re
     // don't remember any context
     resetContext();
 
+    processSystemPrompt();
+
```

Comment:
processSystemPrompt already implies resetContext, but loadModel already implies processSystemPrompt if it succeeds. I believe these two lines can be removed to save unnecessary work, because the m_isServer special-case was removed.

Reply:
No, this is necessary. The server option needs to reprocess the system prompt on every call to completion. ChatLLM::prompt(...) calls processSystemPrompt, but the server does not use this... why? Because it needs to be able to pass in its own generation settings so it calls promptInternal for this very reason.

Reply:
Fixed by calling processSystemPrompt() from promptInternal as suggested.